src/types.rs

```diff
@@ -268,24 +276,103 @@ impl Type {
         TypeBound::Copyable.contains(self.least_upper_bound())
     }
 
+    /// Checks all variables used in the type are in the provided list
+    /// of bound variables, and that for each [CustomType] the corresponding
+    /// [TypeDef] is in the [ExtensionRegistry] and the type arguments
+    /// [validate] and fit into the def's declared parameters.
+    ///
+    /// [validate]: crate::types::type_param::TypeArg::validate
+    /// [TypeDef]: crate::extension::TypeDef
     pub(crate) fn validate(
         &self,
         extension_registry: &ExtensionRegistry,
+        type_vars: &[TypeParam],
```

Comment:
Good point. It's meant to be the declarations, i.e. an explanation of what each variable is, rather than values of the variables. I'll rename to `var_decls` (? - could be `declared_variables` or `bound_vars` or `vars_in_scope` ?)

Reply:
I like `var_decls` 👍